integrations/LaunchDarkly/browser.js

```diff
@@ -0,0 +1,84 @@
+import {
+    isArray,
+    isObject,
+} from "../../utils/utils"; import logger from "../../utils/logUtil";
+import ScriptLoader from "../ScriptLoader";
+const isString = function isString(val) {
+    return toStr$1.call(val) === "[object String]";
+};
+class LaunchDarkly {
+    constructor(config) {
+        this.name = "LaunchDarkly";
+        this.clientSideId = config.clientSideId;
+    }
+    init() {
+        logger.debug("===in init LaunchDarkly===");
+        if (!this.clientSideId) {
+            logger.debug("clientSideId is missing");
+            return;
+        }
+        ScriptLoader(null, "https://unpkg.com/launchdarkly-js-client-sdk@2");
+    }
+    isLoaded() {
+        logger.debug("===In isLoaded LaunchDarkly===");
+        return !!window.LDClient;
+    }
+    isReady() {
+        logger.debug("===In isReady LaunchDarkly===");
+        if (window.LDClient)
+            return true;
+        else
+            return false;
+    }
+    identify(rudderElement) {
+        var user = {};
+        const { message } = rudderElement;
+        user.key = message.userId || message.anonymousId;
+        const { traits } = message.context;
+        if (traits.anonymous !== undefined)
+            user.anonymous = traits.anonymous;
+        if (traits.avatar !== undefined)
+            user.avatar = traits.avatar
+        if (traits.country != undefined)
+            user.country = traits.country
+        if (traits.custom != undefined && isObject(traits.custom))
+            user.custom = traits.custom;
+        if (traits.email != undefined)
+            user.email = traits.email;
+        if (traits.ip != undefined && isString(traits.ip))
+            user.ip = traits.ip;
+        if (traits.lastName != undefined && isString(traits.lastName))
+            user.lastName = traits.lastName;
+        if (traits.name != undefined && isString(traits.name))
+            user.name = traits.name;
+        if (traits.privateAttributeNames != undefined && isArray(traits.privateAttributeNames))
+            user.privateAttributeNames = traits.privateAttributeNames;
+        if (traits.secondary != undefined && isString(traits.secondary))
+            user.secondary = traits.secondary;
```

Comment:
Do we need this extensive validation?

Reply:
From the destination, all these datatypes are mentioned. So I am checking.

Reply:
We should move this huge list of validations to util file and wrap it in separate function like `fieldValidator()` . It will make browser file clean and more readable.

Reply:
Okay, I can create a separate function in util that will build the user object.

Reply:
What happens if say `user.email` is set to `undefined`?

Reply:
Then I am not sending email in user object
.

Reply:
In that case, we can skip validation for some of the parameters as LD is already taking care of them.

Reply:
Okay

Reply:
Along the lines of `undefined` checks, are `isString` validations required?

Reply:
Okay, will change them.